icicle/tests/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+cmake_minimum_required(VERSION 3.18)
+
+include(../cmake/Common.cmake)
+include(../cmake/CurvesCommon.cmake)
+include(../cmake/FieldsCommon.cmake)
+
+set_env()
+set_gpu_env()
+
+project(icicle_tests LANGUAGES CUDA CXX)
```

Comment:
I have a concern here: polynomial test in my branch links to icicle but I don't see how I will be able to do it this way. Same for benchmarks.
I think this is solved by not making tests a standalone project but call add_subdirectory() from the top level cmake so that the test target can link to icicle_curve or icicle_field.

Reply:
sure, we can do that